version: '3'

services:
  lumimoney-app:
    image: lumimoney-app
    pull_policy: never
    restart: always
    container_name: lumimoney-app
    environment:
      API_BASE_URL: https://api.lumimoney.thiagosol.com
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.lumimoney.entrypoints=http"
      - "traefik.http.routers.lumimoney.rule=Host(`lumimoney.thiagosol.com`)"
      - "traefik.http.routers.lumimoney-secure.entrypoints=https"
      - "traefik.http.routers.lumimoney-secure.rule=Host(`lumimoney.thiagosol.com`)"
      - "traefik.http.routers.lumimoney-secure.tls=true"
      - "traefik.http.routers.lumimoney-secure.service=lumimoney-secure"
      - "traefik.http.services.lumimoney-secure.loadbalancer.server.port=80"
      - "traefik.docker.network=external-sol-apis"
    networks:
      - external-sol-apis
      - logging-network
    logging:
      driver: loki
      options:
        loki-url: "http://loki:3100/loki/api/v1/push"
        loki-external-labels: "service=lumimoney-app,type=service"
        loki-retries: "5"
        loki-batch-size: "400"
        mode: "non-blocking"
    deploy:
      resources:
        limits:
          memory: 100M

networks:
  logging-network:
    external: true
  external-sol-apis:
    external: true
